src/validation.cpp

```diff
@@ -1885,6 +1887,13 @@ DisconnectResult CChainState::DisconnectBlock(const CBlock& block, const CBlockI
         }
     }
 
+    if (fTimestampIndex) {
+        if (!pblocktree->EraseTimestampIndex(CTimestampIndexKey(pindex->nTime, pindex->GetBlockHash()))) {
+            AbortNode("Failed to delete timestamp index");
```

Comment:
can we add additional information here such as the timestamp and block hash that we tried to erase?

Reply:
That's just one of many `AbortNodes` for these indexes, I'd keep them unified. Also, `DisconnectBlock` callers log enough info already imo.